nemo/lightning/pytorch/strategies/megatron_strategy.py

```diff
@@ -652,17 +652,19 @@ def load_model_state_dict(self, checkpoint: Mapping[str, Any], strict: bool = Tr
     @property
     @override
     def checkpoint_io(self) -> CheckpointIO:
-        return get_checkpoint_io(
-            self._checkpoint_io,
```

Comment:
A bit confused why self._checkpoint_io reference is not being passed.

Reply:
no, i meant before the PR, why doesn't `self._checkpoint_io` get set ? If it's passed by reference, it should be set ? We observed that it wasn't being set by `get_checkpoint_io`, but I don't understand why.

Reply:
ok, makes sense. good to know